### PR TITLE
ORC-754: Code cleanup

### PR DESCRIPTION
**Pull request by pavibhai**
### What changes were proposed in this pull request?

* Removal of group name in bench pom.xml
* Replaced Charset.forName with StandardCharsets
* Final on some variables
* Removed final on a private method
* Simplified some conditions
* Removed redundant casts
* Removed duplicate condition branches

### Why are the changes needed?
Code cleanup

### How was this patch tested?
As this does not change any of the existing functionality. It was verified that none of the existing tests break.

### CODE REVIEW

**Review by omalley — CHANGES_REQUESTED** [review]
This is a good set of changes, but I don't think we need the bench pom change.

File: java/bench/core/pom.xml [hunk 1]
```diff
@@ -116,6 +115,10 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>8</source>
+          <target>8</target>
+        </configuration>
```
[the remark this answers is not in the document]
Comment (pavibhai) [reply java/bench/core/pom.xml:121]: agreed

**Comment by omalley** [comment]
I just committed this. Thanks, Pavan!

**Comment by dongjoon-hyun** [comment]
Thank you, @pavibhai and @omalley .

**Pull request merged**